oak-store-document/src/test/java/org/apache/jackrabbit/oak/plugins/document/VersionGCTest.java

```diff
@@ -531,6 +531,51 @@ public void testResetFullGCDryRunMode() throws Exception {
 
     // OAK-10370 END
 
+    // OAK-10896
+
+    @Test
+    public void testVersionGCLoadGCModeConfigurationNotApplicable() throws Exception {
+        int FULL_GC_MODE_NOT_ALLOWED_VALUE = 5;
+        VersionGarbageCollector gc = new VersionGarbageCollector(
+                ns, new VersionGCSupport(store), true, false, false,
+                FULL_GC_MODE_NOT_ALLOWED_VALUE);
+
+        assertEquals("Starting VersionGarbageCollector with not applicable / not allowed value" +
+                "will set fullGcMode to default NONE", gc.getFullGcMode(), VersionGarbageCollector.FullGCMode.NONE);
```

Comment:
this test should actually have failed,a s the coded default `VersionGarbageCollector.FullGCMode` is `GAP_ORPHANS_EMPTYPROPS` - the test only succeeds as a side-effect of `setUp` building a `DocumentNodeStore` which uses `DocumentNodeStoreBuilder.getFullGCMode` which uses an uninitialized thus default value for `fullGCMode`. 

I' suggest to change the test to first set a valid mode (eg 2) an then set an invalid mode (eg the 5) and then it should have been set to 0

Reply:
I updated the test, thank you for the suggestion.